pkg/deck/deck.go

```diff
@@ -0,0 +1,173 @@
+package deck
+
+import (
+	"math/rand"
+	"slices"
+)
+
+// Deck represents a deck of cards.
+type Deck struct {
+	// Cards represents the cards in the deck.
+	Cards []Card
+}
+
+// Config represents the configuration for the deck.
+type Config struct {
+	// cmpFunc is the function used to compare cards for sorting the deck.
+	cmpFunc func(a, b Card) int
+
+	// shuffleDeck indicates whether to shuffle the deck.
+	shuffleDeck bool
+
+	// jokerInDeckCount is the number of jokers in the deck.
+	jokerInDeckCount int
+
+	// excludeFunc is the function used to exclude cards from the deck.
+	excludeFunc func(Card) bool
+
+	// additionalDecksCount is the number of additional decks to create.
+	additionalDecksCount int
+}
+
+// ConfigOption is a function that modifies the Config struct.
+type ConfigOption func(*Config)
+
+// WithJokersInDeck sets the number of jokers in a deck.
+func WithJokersInDeck(jokerCount int) ConfigOption {
+	return func(c *Config) {
+		c.jokerInDeckCount = jokerCount
+	}
+}
+
+// WithCompareFunc sets the comparison function for sorting the deck.
+// If shuffling is enabled as well, the deck will be sorted after shuffling.
+func WithCompareFunc(cmp func(a, b Card) int) ConfigOption {
+	return func(c *Config) {
+		c.cmpFunc = cmp
+	}
+}
+
+// WithShuffle sets whether to shuffle the deck.
+func WithShuffle(shuffle bool) ConfigOption {
+	return func(c *Config) {
+		c.shuffleDeck = shuffle
+	}
+}
+
+// WithExclude sets the function to exclude cards from the deck.
+func WithExclude(exclude func(Card) bool) ConfigOption {
+	return func(c *Config) {
+		c.excludeFunc = exclude
+	}
+}
+
+// WithAdditionalDecks sets the number of additional decks to create.
+func WithAdditionalDecks(additionalDecksCount int) ConfigOption {
+	return func(c *Config) {
+		c.additionalDecksCount = additionalDecksCount
+	}
+}
+
+// NewDeck creates a new deck with the given options.
+func NewDeck(opts ...ConfigOption) *Deck {
+	cfg := Config{}
+	for _, o := range opts {
+		o(&cfg)
+	}
+
+	deck := buildDecks(cfg)
+
+	deck = modifyDeckContent(cfg, deck)
+
+	deck = modifyDeckOrder(cfg, deck)
+
+	return deck
+}
+
+// buildDecks creates a new deck with the given configuration.
```

Comment:
Since go doc comments use complete sentences, it should start with a capital letter. Also it's not mandatory to mention the func name because it's not exported. 

```suggestion
// The func creates a new deck with the given configuration.
```

```suggestion
// The buildDecks func creates a new deck with the given configuration.
```

Reply:
The Golang style guide provides an exception for symbols that begin with an uncapitalized identifier name (https://google.github.io/styleguide/go/decisions.html#comment-sentences)

The Golang standart library also does this:
* https://github.com/golang/go/blob/master/src/bytes/bytes.go#L32
* https://github.com/golang/go/blob/master/src/bytes/bytes.go#L32